### PR TITLE
Add memory usage dashboard for notebook and dask-gateway pods

### PR DESCRIPTION
**Pull request by GeorgianaElena**
This adds a dashboard to track pod memory requests across different type of pods, grouped by username (the usernames are not present in the screenshots intentionally, to keep them secret)

1. notebook pods
Screenshot 2022-06-15 at 10 58 10

2. dask-worker pods
Screenshot 2022-06-15 at 11 01 03

3. dask-scheduler pods
Screenshot 2022-06-15 at 11 01 15

Reference https://github.com/2i2c-org/infrastructure/issues/1279

### CODE REVIEW

File: dashboards/usage-report.jsonnet [hunk 1]
```diff
@@ -57,7 +53,45 @@ local memoryUsageUserPods = barGaugePanel.new(
...
+  'Dask-gateway worker pod memory usage',
+  datasource='$PROMETHEUS_DS',
+  unit='bytes',
+  thresholds=[
+    {
+      value: 0,
+      color: 'green',
+    },
+    {
+      value: 600,
```
Comment (yuvipanda) [inline dashboards/usage-report.jsonnet:73]: Let's leave these thresholds empty for now? 600 is a bit of an arbitrary value, right?

**Review by yuvipanda — APPROVED** [review]
Let's get rid of thresholds for now and we can revisit later? Otherwise LGTM to deploy!

File: dashboards/usage-report.jsonnet (hunk 1 above)
Comment (GeorgianaElena) [reply dashboards/usage-report.jsonnet:73]: yes, mostly arbitrarily. I added the thresholds to make the dashboard easier to read. No thresholds will produce a dashboard like the one in the initial comment (all grey, hard to distinguish what's what) and this is a dashboard with thresholds:
Screenshot 2022-06-15 at 09 41 12


We could keep only the base threshold? and have the bars be one color?

Comment (GeorgianaElena) [reply dashboards/usage-report.jsonnet:73]: uuuh, I like it on pink also

Screenshot 2022-06-15 at 10 39 32

(*ups, I think I'm down a rabbit hole and now I should probably stop playing with colors* 😅 )

**Comment by GeorgianaElena** [comment]
> Otherwise LGTM to deploy!

@yuvipanda, I'm a bit confused. Does this mean that I should merge it too?
Btw, this is already deployed to the 2i2c central grafana. For example https://grafana.pilot.2i2c.cloud/d/usage-report/usage-report?orgId=1&from=now-7d&to=now&var-PROMETHEUS_DS=leap&var-hub=All if you want to check it out.

**Comment by GeorgianaElena** [comment]
update: last commit adds a dashboard for gpu pods

**Comment by yuvipanda** [comment]
Ah, thanks @GeorgianaElena! This looks great!

**Pull request merged**